hypothesis-python/src/hypothesis/internal/scrutineer.py

```diff
@@ -38,11 +49,36 @@ def trace(self, frame, event, arg):
         if event == "call":
             return self.trace
         elif event == "line":
-            fname = frame.f_code.co_filename
-            if should_trace_file(fname):
-                current_location = (fname, frame.f_lineno)
-                self.branches.add((self._previous_location, current_location))
-                self._previous_location = current_location
+            self.trace_line(frame.f_code, frame.f_lineno)
```

Comment:
Let's keep this inline - the performance overhead of a function call in Python is very small, but doing so for _every line of code_ adds up and it's worth duplicating a few lines of code to avoid that.

Reply:
good point!